lambda.tf

```diff
@@ -23,15 +23,9 @@ resource "aws_lambda_function" "lambda" {
   filename   = "${lookup(data.external.built.result, "filename")}"
   depends_on = ["null_resource.archive"]
 
-  # The aws_lambda_function resource has a schema for the environment
-  # variable, where the only acceptable values are:
-  #   a. Undefined
-  #   b. An empty list
-  #   c. A list containing 1 element: a map with a specific schema
-  # Use slice to get option "b" or "c" depending on whether a non-empty
-  # value was passed into this module.
-
-  environment = ["${slice( list(var.environment), 0, length(var.environment) == 0 ? 0 : 1 )}"]
+  environment {
+    variables = var.environment.variables
+  }
```

Comment:
I honestly didn't know what this was doing, or what the correct change would be.
This seems to work for me (probably because I'm indeed passing variables, and not an empty value, which seems to be the edge case covered here), so I went with it for lack of a better short-term solution